<h2>Why is this an issue?</h2>
<p><code>Exit Function</code>, <code>Exit Property</code>, and <code>Exit Sub</code> are all poor, less-readable substitutes for a simple
<code>Return</code>, and if used with code that should return a value (<code>Exit Function</code> and in some cases <code>Exit Property</code>) they
could result in a <code>NullReferenceException</code>.</p>
<p>This rule raises an issue for all their usages.</p>
<h3>Noncompliant code example</h3>
<pre>
Public Class Sample

  Public Sub MySub(Condition As Boolean)
    If Condition Then Exit Sub                  ' Noncompliant
    ' ...
  End Sub

  Public Function MyFunction(Condition As Boolean) As Integer
    If Condition Then
        MyFunction = 42
        Exit Function              ' Noncompliant
    End If
    ' ...
  End Function

End Class
</pre>
<h3>Compliant solution</h3>
<pre>
Public Class Sample

  Public Sub MySub(Condition As Boolean)
    If Condition Then Return                  ' Noncompliant
    ' ...
  End Sub

  Public Function MyFunction(Condition As Boolean) As Integer
    If Condition Then Return 42
    ' ...
  End Function

End Class
</pre>

